### PR TITLE
feat: save semantic viewer charts and add in the dashboards

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: #11366 
Closes: https://github.com/lightdash/lightdash/issues/11368

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/backend/src/analytics/LightdashAnalytics.ts [hunk 1]
```diff
@@ -740,6 +741,32 @@ export type CreateSqlChartVersionEvent = BaseTrack & {
...
+    event: 'semantic_layer_chart_version.created';
+    userId: string;
+    properties: {
+        chartId: string;
+        versionId: string;
+        projectId: string;
+        organizationId: string;
+        chartKind: ChartKind;
+        semanticLayerQuery: SemanticLayerQuery;
+        barChart?: {
```
Comment (IrakliJani) [inline packages/backend/src/analytics/LightdashAnalytics.ts:754]: this does not have table config. do we also need to keep all configuration blobs as the analytics event or the active one @TuringLovesDeathMetal

File: packages/backend/src/services/SavedSemanticLayerService/SavedSemanticLayerService.ts [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+        semanticLayerChart: SemanticLayerCreateChart,
+    ): Promise<ApiSemanticLayerCreateChart['results']> {
+        const { organizationUuid } = await this.projectModel.getSummary(
+            projectUuid,
+        );
+        if (
+            user.ability.cannot(
+                'manage',
+                // TODO: add it's own ability
+                subject('CustomSql', { organizationUuid, projectUuid }),
```
Comment (IrakliJani) [inline packages/backend/src/services/SavedSemanticLayerService/SavedSemanticLayerService.ts:206]: add `ability`

File: packages/frontend/src/components/DashboardTiles/DashboardSemanticLayerChartTile.tsx [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+    const resultsRunner = useMemo(() => {
+        if (!data || !fields) return;
+
+        return new SemanticViewerResultsRunner({
+            projectUuid,
+            fields,
+            query: data.chart.semanticLayerQuery,
+            rows: chartData.results,
+            columns: [], // TODO: sqlRunnerChartData.columns,
```
Comment (IrakliJani) [inline packages/frontend/src/components/DashboardTiles/DashboardSemanticLayerChartTile.tsx:85]: forgot column conversion here

File: packages/frontend/src/components/common/Page/Sidebar.tsx [hunk 4]
```diff
@@ -107,20 +101,21 @@ const Sidebar: FC<React.PropsWithChildren<Props>> = ({
                 >
                     {(style) => (
                         <>
-                            <Card
```
Comment (IrakliJani) [inline packages/frontend/src/components/common/Page/Sidebar.tsx:110]: card is too opinionated to be a sidebar component

File: packages/backend/src/services/SavedSemanticLayerService/SavedSemanticLayerService.ts (hunk 2 above)
Comment (IrakliJani) [reply packages/backend/src/services/SavedSemanticLayerService/SavedSemanticLayerService.ts:206]: can be done as a separate ticket

File: packages/backend/src/analytics/LightdashAnalytics.ts (hunk 1 above)
Comment (IrakliJani) [reply packages/backend/src/analytics/LightdashAnalytics.ts:754]: @TuringLovesDeathMetalsSorry for providing less than a zero context 😂 

I noticed that with SQL Runner, when you save a chart, we fire an analytics event. The analytics event includes all of the configurations for all charts, not just the active chart that's being used. Additionally, we are not including the table visualization configuration at all. I copied this "blindly" from SQL Runner but left a comment not to forget to update both places if needed.

Comment (TuringLovesDeathMetal) [reply packages/backend/src/analytics/LightdashAnalytics.ts:754]: >The analytics event includes all of the configurations for all charts, not just the active chart that's being used.


wut - that sounds wild 😂 

Ya, can we _**just**_ include the active chart configuration. Also, if there's a table visualization, then include it's configuration

**Review by almeidabbm — APPROVED** [review]

**Pull request merged**